asapdiscovery-docking/asapdiscovery/docking/schema/pose_generation.py

```diff
@@ -293,115 +452,280 @@ def _generate_poses(
 
         return posed_ligands, failed_ligands
 
-    def _prune_clashes(self, receptor: oechem.OEMol, ligands: list[oechem.OEMol]):
+
+class RDKitConstrainedPoseGenerator(_BasicConstrainedPoseGenerator):
+    """Use RDKit to embed multiple conformers of the molecule while constraining it to the template ligand."""
+
+    type: Literal["RDKitConstrainedPoseGenerator"] = "RDKitConstrainedPoseGenerator"
+
+    max_confs: PositiveInt = Field(
+        300, description="The maximum number of conformers to try and generate."
+    )
+    rms_thresh: PositiveFloat = Field(
+        0.2,
+        description="Retain only the conformations out of 'numConfs' after embedding that are at least this far apart from each other. RMSD is computed on the heavy atoms.",
+    )
+    mcs_timeout: PositiveFloat = Field(
```

Comment:
I'm just checking with OpenFE folks if they have a recommended default value here.

Reply:
Good idea 1 seconds seems enough most of the time but would be good to check!

Reply:
okay, I was trying to set <1 but that fails. Looking a cryptic `_pickle.PicklingError`. Looking at the RDKit docs, seems like it can only be `int`, so 1 is the minimum I think. I remember peers using `timeout = 1` now that I think about it some more, so let's just stick with that. Would be good to at least set `PositiveFloat` -> `PositiveInt`?

Reply:
Good spot Ill update that.